lib/trino-parquet/src/main/java/io/trino/parquet/reader/BitPackingUtils.java

```diff
@@ -0,0 +1,236 @@
+        values[offset + 31] = ((packedValue >>> 31) & 1) == 1;
+        values[offset + 32] = ((packedValue >>> 32) & 1) == 1;
+        values[offset + 33] = ((packedValue >>> 33) & 1) == 1;
+        values[offset + 34] = ((packedValue >>> 34) & 1) == 1;
+        values[offset + 35] = ((packedValue >>> 35) & 1) == 1;
+        values[offset + 36] = ((packedValue >>> 36) & 1) == 1;
+        values[offset + 37] = ((packedValue >>> 37) & 1) == 1;
+        values[offset + 38] = ((packedValue >>> 38) & 1) == 1;
+        values[offset + 39] = ((packedValue >>> 39) & 1) == 1;
+
+        values[offset + 40] = ((packedValue >>> 40) & 1) == 1;
+        values[offset + 41] = ((packedValue >>> 41) & 1) == 1;
+        values[offset + 42] = ((packedValue >>> 42) & 1) == 1;
+        values[offset + 43] = ((packedValue >>> 43) & 1) == 1;
+        values[offset + 44] = ((packedValue >>> 44) & 1) == 1;
+        values[offset + 45] = ((packedValue >>> 45) & 1) == 1;
+        values[offset + 46] = ((packedValue >>> 46) & 1) == 1;
+        values[offset + 47] = ((packedValue >>> 47) & 1) == 1;
+        values[offset + 48] = ((packedValue >>> 48) & 1) == 1;
+        values[offset + 49] = ((packedValue >>> 49) & 1) == 1;
+
+        values[offset + 50] = ((packedValue >>> 50) & 1) == 1;
+        values[offset + 51] = ((packedValue >>> 51) & 1) == 1;
+        values[offset + 52] = ((packedValue >>> 52) & 1) == 1;
+        values[offset + 53] = ((packedValue >>> 53) & 1) == 1;
+        values[offset + 54] = ((packedValue >>> 54) & 1) == 1;
+        values[offset + 55] = ((packedValue >>> 55) & 1) == 1;
+        values[offset + 56] = ((packedValue >>> 56) & 1) == 1;
+        values[offset + 57] = ((packedValue >>> 57) & 1) == 1;
+        values[offset + 58] = ((packedValue >>> 58) & 1) == 1;
+        values[offset + 59] = ((packedValue >>> 59) & 1) == 1;
+
+        values[offset + 60] = ((packedValue >>> 60) & 1) == 1;
+        values[offset + 61] = ((packedValue >>> 61) & 1) == 1;
+        values[offset + 62] = ((packedValue >>> 62) & 1) == 1;
+        values[offset + 63] = ((packedValue >>> 63) & 1) == 1;
+
+        return Long.SIZE - Long.bitCount(packedValue);
+    }
+
+    public static void unpack(byte[] values, int offset, byte packedByte)
+    {
+        values[offset] = (byte) (packedByte & 1);
+        values[offset + 1] = (byte) ((packedByte >>> 1) & 1);
+        values[offset + 2] = (byte) ((packedByte >>> 2) & 1);
+        values[offset + 3] = (byte) ((packedByte >>> 3) & 1);
+        values[offset + 4] = (byte) ((packedByte >>> 4) & 1);
+        values[offset + 5] = (byte) ((packedByte >>> 5) & 1);
+        values[offset + 6] = (byte) ((packedByte >>> 6) & 1);
+        values[offset + 7] = (byte) ((packedByte >>> 7) & 1);
+    }
+
+    public static void unpack(byte[] values, int offset, long packedValue)
+    {
+        values[offset] = (byte) (packedValue & 1);
+        values[offset + 1] = (byte) ((packedValue >>> 1) & 1);
```

Comment:
Are we sure this is better than a loop? What does the resulting assembly and perf look for this? How did we measure it? As in the comment above, the downside is more bytecode and fewer opportunities for inlining as a result.

Reply:
Those unrollings have show significant improvements in jmh benchmarks and some visible (not huge obviously) gains in macrobenchmarks.